### ISSUE TITLE
ViTPose+ models post processing doest not work for `dataset_index : 5`

### ISSUE THREAD

**Issue by testdummyvt** [huggingface/transformers#39789]
### System Info

- `transformers` version: 4.54.1
- Platform: macOS-15.1-arm64-arm-64bit
- Python version: 3.11.12
- Huggingface_hub version: 0.34.3
- Safetensors version: 0.5.3
- Accelerate version: 1.7.0
- Accelerate config: 	not found
- DeepSpeed version: not installed
- PyTorch version (accelerator?): 2.7.1 (NA)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@qubvel 

When I change the `inputs["dataset_index"] = torch.tensor([0], device=device)` to `inputs["dataset_index"] = torch.tensor([5], device=device)`

The post-processing fails to process the keypoints correctly.

When I keep `dataset_index = 0`, I get the following results:  

<img width="640" height="426" alt="Image" src="https://github.com/user-attachments/assets/e1d3dbe8-f3f7-4340-ab98-b5c13a03fbfc" />

But when I set `dataset_index = 5`, I get the following results:  

<img width="640" height="426" alt="Image" src="https://github.com/user-attachments/assets/41efd8c1-8a28-4bc1-8610-5826266f6e50" />

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

[From Usage Tips](https://huggingface.co/docs/transformers/en/model_doc/vitpose#usage-tips)

I added `inputs["dataset_index"] = torch.tensor([5], device=device)` in the official example just after 
`inputs = image_processor(image, boxes=[person_boxes], return_tensors="pt").to(device)` line.

### Expected behavior

With dataset_index = 5, which is the cocowholebody dataset, we need to get `133` keypoints.

**Comment by qubvel** [comment]
cc @SangbumChoi if you have any thoughts re the issue

**Comment by SangbumChoi** [comment]
@testdummyvt @qubvel AFAIK, we have only implemented the case of dataset_index : 0 in post-processing

**Comment by SangbumChoi** [comment]
https://github.com/ViTAE-Transformer/ViTPose/blob/d5216452796c90c6bc29f5c5ec0bdba94366768a/mmpose/models/detectors/top_down_moe.py#L78

As you can see the associated keypoint head is kind of working as auxiliuary loss so I think it requires additional work to make this feature compatible.

**Comment by testdummyvt** [comment]
@SangbumChoi  I see it's good to know. 👍 

It would be nice if we could get a warning when we change dataset_index from 0.
